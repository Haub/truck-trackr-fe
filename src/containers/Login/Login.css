.login {
  margin-top: 110px;
  position: relative;
  background-color: #FEECC8;
  border: 2px solid #6D3E48;
  border-radius: 1rem;
  box-shadow: 0px 1px 3px 2px #0000005c;
}

form {
  height: 24rem;
  width: 20rem;
  display: flex;
  align-items: center;
  justify-content: space-around;
  flex-direction: column;
}

main {
  display: flex;
  align-items: center;
  justify-content: center;
  flex-direction: column;
  width: 44vw;
  height: 33rem;
}

.button-holder {
  display: flex;

}

.email-input,
.password-input,
.business-name-input,
.address-input,
.phone-number-input,
.contact-name-input,
.food-type-input {
  background-color: #fef9ee;
  border: 1px solid #6D3E48;
  display: flex;
  width: 19rem;
  align-content: center;
  height: 2rem;
  margin-top: 1rem;
  border-radius: 5px;
  padding-left: 5px;
  font-size: 1.3rem;
  opacity: 1;
}

.signin-button, .signup-button {
  color: white;
  background-color: #6D3E48;
  width: 19rem;
  height: 2rem;
  border: none;
  border-radius: 5px;
  font-size: 1.1rem;
  margin-top: 2rem;
  margin-bottom: 25px;
}
.email-input::placeholder,
.name-input::placeholder,
.password-input::placeholder {
  color: #8C8C8C;
}

.logo-input {
  margin-top: 1.3rem;
}

.logo-button {
  width: 2rem;
}

.hidden {
  display: none;
  transition: all .2s;
}