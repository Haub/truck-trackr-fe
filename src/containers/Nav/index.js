import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import './nav.css'

export const Nav = (props) => {

    return (
        <nav>
            
        </nav>
    )
}