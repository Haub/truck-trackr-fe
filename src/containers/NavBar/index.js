import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { loadFoodTrucks, loadBreweries, loadProfile } from '../../actions';
import { mockFoodTrucks, mockBreweries } from '../../tests/testMocks';

import './NavBar.css'

export class NavBar extends Component {
    constructor() {
        super()
        this.state = {

        }
    }

    loadData = async (event) => {
        const businessType = event.target.getAttribute('name')

        try {
            const response = await fetch(`https://truck-trackr-api.herokuapp.com/api/v1/${businessType}`);
            const result = await response.json();
            if(businessType === 'breweries'){
                this.props.loadProfile({page: 'breweries'})
                this.props.loadBreweries(result.data)
            } else {
                this.props.loadProfile({page: 'food_trucks'})
                this.props.loadFoodTrucks(result.data)
            }
        } catch(error) {
            throw new Error(error.message)
        }
    }

    loadMockData = (event) => {
        const businessType = event.target.getAttribute('name')

        if(businessType === 'breweries') {
            this.props.loadBreweries(mockBreweries)
        } else if(businessType === 'food_trucks') {
            this.props.loadFoodTrucks(mockFoodTrucks)
        }
    }

    render(){
        return (
            <nav>
                <ul className='nav-controls'>
                    <li className='nav-control my-profile'>My Profile</li>
                    <NavLink to='/breweries' onClick={this.loadData} name='breweries' className='nav-control breweries'>Breweries</NavLink>
                    <NavLink to='/food_trucks' onClick={this.loadData} name='food_trucks' className='nav-control food-trucks'>Food Trucks</NavLink>
                    <input className='search-input' placeholder='Search'/>
                </ul>
            </nav>
        )
    }
}

export const mapDispatchToProps = (dispatch) => ({
    loadBreweries: (breweries) => dispatch(loadBreweries(breweries)),
    loadFoodTrucks: (foodTrucks) => dispatch(loadFoodTrucks(foodTrucks)) ,
    loadProfile: (profile) => dispatch(loadProfile(profile))
});

export default connect(null, mapDispatchToProps)(NavBar);