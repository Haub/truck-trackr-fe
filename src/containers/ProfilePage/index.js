import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import "./ProfilePage.css";
import SignUpPage from "../../components/SignUpPage";
import { addUser } from "../../actions";
import { createEventFetch } from "../../utilities";

export class ProfilePage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      createEventOpen: false,
      eventName: "",
      eventDate: ""
    };
  }

  componentDidMount() {
    // if(!Object.keys(this.props.currentPage).length){
    //     this.props.history.push('/')
    // }
  }

  makeEvent = () => {
    let newState = !this.state.createEventOpen;
    this.setState({ createEventOpen: newState });
  };

  handleKeyPress = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async e => {
    e.preventDefault();
    let cleanUser = this.cleanDataForRequest();
    let cleanPostParams = this.createPostBody();
    let result = await createEventFetch(cleanUser, cleanPostParams);
    console.log(result);
    this.setState({
      createEventOpen: false,
      eventName: "",
      eventDate: ""
    });
  };

  cleanDataForRequest = () => {
    const { user } = this.props;
    let type;
    let eventType;
    if (user.type === "food_truck") {
      type = "food_trucks";
      eventType = "openDates";
    } else {
      type = "breweries";
      eventType = "brewery_events";
    }

    let cleanUser = {
      type,
      id: user.id,
      eventType
    };
    return cleanUser;
  };

  createPostBody = () => {
    const uid = sessionStorage.getItem("uid");
    const { eventDate } = this.state;

    return {
      uid,
      date: eventDate
    };
  };

  render() {
    const { currentPage, user } = this.props;

    if (Object.keys(currentPage).length) {
      const { attributes } = currentPage;
      return (
        <div className="parent-container">
          <div className="profile-container">
            <div className="biz-container">
              <h2 className="biz-name">{attributes.name}</h2>
              <h5
                className={
                  currentPage.type === "food_truck" ? "biz-title" : "hidden"
                }
              >
                Food Type:
              </h5>
              <h5
                className={
                  currentPage.type === "food_truck" ? "biz-info" : "hidden"
                }
              >
                {attributes.food_type || null}
              </h5>

              <h5 className="biz-title">Phone Number:</h5>
              <h5 className="biz-info">{attributes.phone}</h5>

              <h5 className="biz-title">Contact Name:</h5>
              <h5 className="biz-info">{attributes.contact_name}</h5>

              <h5 className="biz-title">Email:</h5>
              <h5 className="biz-info">{attributes.email}</h5>

              <h5 className="biz-title">Website:</h5>
              <a href={attributes.website} target="blank" className="biz-info">
                {attributes.website}
              </a>
            </div>
            <button
              onClick={this.makeEvent}
              className={
                currentPage === user ? "create-event-button" : "hidden"
              }
            >
              Create Event
            </button>
            <div className="upcoming-events-container">
              <h3 className="upcoming-events-title">UPCOMING EVENTS</h3>
              <h4 className="upcoming-events-date">
                12/12{" "}
                <span className="upcoming-events-status">Need a Booking</span>
              </h4>
              <h4 className="upcoming-events-date">
                12/13 <span className="upcoming-events-status">@ Cerebral</span>
              </h4>
              <h4 className="upcoming-events-date">
                12/14 <span className="upcoming-events-status">@ Diebolt </span>
              </h4>
              <h4 className="upcoming-events-date">
                12/15{" "}
                <span className="upcoming-events-status">Need a Booking</span>
              </h4>
              <h4 className="upcoming-events-date">
                12/16{" "}
                <span className="upcoming-events-status">@ Call To Arms</span>
              </h4>
              <h4 className="upcoming-events-date">
                12/17{" "}
                <span className="upcoming-events-status">
                  @ Zuni Street Brewery{" "}
                </span>
              </h4>
              <h4 className="upcoming-events-date">
                12/18{" "}
                <span className="upcoming-events-status">Need a Booking</span>
              </h4>
            </div>
            <form
              className={
                this.state.createEventOpen ? "create-event-form" : "hidden"
              }
            >
              <input
                className="create-event-input"
                name="eventName"
                value={this.state.eventName}
                placeholder="Event Name"
                onChange={this.handleKeyPress}
              />
              <input
                className="create-event-input"
                name="eventDate"
                value={this.state.eventDate}
                placeholder="Event Date (numerical)"
                onChange={this.handleKeyPress}
              />
              <button
                onClick={this.handleSubmit}
                className="create-event-form-button"
              >
                Create
              </button>
            </form>
          </div>
        </div>
      );
    } else {
      return <div />;
    }
  }
}

export const mapStateToProps = state => ({
  currentPage: state.currentPage,
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(addUser(user))
});

const { object, func } = PropTypes;

ProfilePage.propTypes = {
  user: object,
  currentPage: object,
  addUser: func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(ProfilePage);
