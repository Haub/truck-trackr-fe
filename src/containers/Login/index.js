import React, { Component } from "react";
import { NavLink, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import "./Login.css";
import { addUser, loadProfile, loadEvents } from "../../actions";
import * as helper from "../../utilities";

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: "",
      passwordOne: "",
      signUp: false,
      locationType: "",
      businessName: "",
      address: "",
      phoneNumber: "",
      contactName: "",
      foodType: "",
      logo: null
    };
  }

  handleKeyPress = event => {
    const { name, value } = event.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async e => {
    // e.preventDefault();
    const { email, passwordOne, signUp, locationType } = this.state;


    if (signUp) {
      this.props.firebase
        .doCreateUserWithEmailAndPassword(email, passwordOne)
        .then(async authUser => {
          let user = this.cleanUserForSignUp(authUser);
          sessionStorage.setItem("uid", authUser.user.uid);
          let result = await helper.createNewUser(user, locationType);
          this.props.addUser(result[0].data);
          this.props.loadProfile(result[0]);
          this.props.loadEvents(result[1].included)
          this.props.history.push("/profile");
        })
        .catch(error => {
          this.setState({ error });
        });
    } else {
      this.props.firebase
        .doSignInWithEmailAndPassword(email, passwordOne)
        .then(async authUser => {
          sessionStorage.setItem("uid", authUser.user.uid);
          const user = this.cleanUserForLogin(authUser, locationType);
          const result = await helper.loginUser(user);
          console.log(result);
          this.props.addUser(result[0].data);
          this.props.loadProfile(result[0]);
          this.props.loadEvents(result[1].included)
          this.props.history.push("/profile");
        })
        .catch(error => {
          this.setState({ error });
        });
    }
  };

  cleanUserForSignUp = authUser => {
    const {
      username,
      email,
      passwordOne,
      signUp,
      locationType,
      businessName,
      address,
      phoneNumber,
      contactName,
      foodType
    } = this.state;
    let user;
    if (locationType === "food_trucks") {
      user = {
        name: businessName,
        food_type: foodType,
        contact_name: contactName,
        phone: phoneNumber,
        email,
        website: username,
        uid: authUser.user.uid
      };
    } else {
      user = {
        name: businessName,
        address: address,
        contact_name: contactName,
        phone: phoneNumber,
        email,
        uid: authUser.user.uid
      };
    }
    return user;
  };

  cleanUserForLogin = (authUser, locationType) => {
    let user;
    if (locationType === "food_trucks") {
      user = {
        uid: authUser.user.uid,
        account_type: "food_truck"
      };
    } else {
      user = {
        uid: authUser.user.uid,
        account_type: "brewery"
      };
    }
    return user;
  };

  toggleSignUp = () => {
    let newSignUp = !this.state.signUp;
    this.setState({ signUp: newSignUp });
  };

  fileChangedHandler = event => {
    this.setState({
      logo: event.target.files[0]
    });
  };

  uploadHandler = () => {
    console.log(this.state.logo);

    // const formData = new FormData()
    // formData.append('myFile', this.state.logo, this.state.logo.name)
    // axios.post('my-domain.com/file-upload', formData)
  };

  render() {
    const { signUp, locationType } = this.state;
    let showFoodType;

    if (signUp && locationType === "food_trucks") {
      showFoodType = true;
    }

    return (
      <main className="login">
        <form className="login-form" onSubmit={this.handleSubmit}>
          <input
            className="email-input"
            name="email"
            type="email"
            placeholder="Email"
            onChange={this.handleKeyPress}
          />
          <input
            className="password-input"
            name="passwordOne"
            type="password"
            placeholder="Password"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "business-name-input" : "hidden"}
            name="businessName"
            placeholder="Business Name"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "address-input" : "hidden"}
            name="address"
            placeholder="Address/City Located"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "phone-number-input" : "hidden"}
            name="phoneNumber"
            placeholder="Phone Number"
            onChange={this.handleKeyPress}
          />
          <input
            className={signUp ? "contact-name-input" : "hidden"}
            name="contactName"
            placeholder="Contact Name"
            onChange={this.handleKeyPress}
          />
          <input
            className={showFoodType ? "food-type-input" : "hidden"}
            name="foodType"
            placeholder="Food Type"
            onChange={this.handleKeyPress}
          />
          <p className={signUp ? "business-type" : "hidden"}>
            What Type of Business:
          </p>
          <div className="button-holder">
            <input
              className={"food-truck-radio"}
              type="radio"
              name="locationType"
              value="food_trucks"
              id="food-truck-button"
              onChange={this.handleKeyPress}
            />
            <label className={"food-truck-label"} htmlFor="food-truck-button">
              Food Truck
            </label>
            <input
              className={"brewery-radio"}
              type="radio"
              name="locationType"
              value="breweries"
              id="brewery-button"
              onChange={this.handleKeyPress}
            />
            <label className={"brewery-label"} htmlFor="brewery-button">
              Brewery
            </label>
          </div>
          {/* <input
            className={signUp ? "logo-input" : "hidden"}
            name="logo"
            placeholder="Upload logo"
            onChange={this.handleKeyPress}
            type="file"
          />
          <button
            className={signUp ? "logo-button" : "hidden"}
            onClick={this.uploadHandler}
          >
            Upload!
          </button> */}
          <NavLink
            to="/profile"
            className="signin-button"
            onClick={e => this.handleSubmit(e)}
          >
            {signUp ? "Sign Up" : "Sign In"}
          </NavLink>
        </form>
        <button className="signup-button" onClick={this.toggleSignUp}>
          {signUp
            ? "Actually I already have an account"
            : "First Time Here? Sign Up!"}
        </button>
      </main>
    );
  }
}

export const mapStateToProps = state => ({
  user: state.user
});

export const mapDispatchToProps = dispatch => ({
  addUser: user => dispatch(addUser(user)),
  loadProfile: currentPage => dispatch(loadProfile(currentPage)),
  loadEvents: events => dispatch(loadEvents(events)) 
});

const { object, func } = PropTypes;

Login.propTypes = {
  user: object,
  addUser: func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(Login));
