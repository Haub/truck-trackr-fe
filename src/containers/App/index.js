import React, { Component } from 'react';
import { Route, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { SignUpPage } from '../../components/SignUpPage';
import { BusinessContainer } from '../../components/BusinessContainer';
import './App.css';
import hamburger from '../../assets/hamburger.svg';
import ProfilePage from '../ProfilePage';
import NavBar from '../NavBar';
import truck from '../../assets/food-truck.png';
import barrel from '../../assets/barrel-icon-new.png';
import { withFirebase } from '../../components/Firebase';
import { loadProfile } from '../../actions'

export class App extends Component {
  constructor() {
    super()
    this.state = {
      login: true,
      navOpen: true,
    }
  }

  componentDidMount(){
    if(!this.state.login){
      this.props.history.push('/login')
    }
    if(!this.props.currentPage) {
      this.props.history.push('/')
    }
  }

  handleNavBar = (e) => {
    document.querySelector('.menu-trigger').classList.toggle('active')
    this.setState ({navOpen: !this.state.navOpen})
  }

  render() {
    let navBar;
    const { navOpen } = this.state;
    const { match } = this.props;

    return (
      <div className='main'>
        <header>
          <h1 className='main-title'><span><img src={ truck } className='truck-icon'/></span>TruckTrackr<span><img src={ barrel } className='barrel-icon'/></span></h1>
          <section>
            <a onClick={ this.handleNavBar} className="menu-trigger" id="menu05">
              <span></span>
              <span></span>
              <span></span>
            </a>
          </section>
          
        </header>

        <div className='content-holder'>
        { this.state.navOpen &&
          <NavBar history={this.props.history}/>
        }
        { !this.state.navOpen &&
          <div id="wrapper">
            <span id="openbtn"  onClick={this.handleNavBar}></span>
          </div>
        }
        <Route
          exact path='/login'
          render={ props => (
            <SignUpPage history={this.props.history}/>
          )
           }
        />
        <Route 
          exact path='/'
          component={ ProfilePage }
        />
        <Route 
          exact path='/breweries'
          render={ props => (
            <BusinessContainer 
              data={this.props.breweries} 
              changeCurrentPage={this.props.changeCurrentPage} 
              currentPage={this.props.currentPage.page}
              loadProfile={this.props.loadProfile}
              history={this.props.history}/>
          )}
        />
        <Route 
          exact path='/food_trucks'
          render={ props => (
            <BusinessContainer 
              data={this.props.foodTrucks} 
              changeCurrentPage={this.props.changeCurrentPage} 
              currentPage={this.props.currentPage.page}
              loadProfile={this.props.loadProfile}
              history={this.props.history}/>
          )}
        />
        <Route path='/business/:businessName' component={ ProfilePage } />
        </div>
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  breweries: state.breweries,
  foodTrucks: state.foodTrucks,
  currentPage: state.currentPage
});

export const mapDispatchToProps = (dispatch) => ({
  loadProfile: (profile) => dispatch(loadProfile(profile))
})

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(withFirebase(App)));