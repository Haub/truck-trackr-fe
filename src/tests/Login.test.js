import React from 'react';
import { shallow, mount } from 'enzyme';
import { Login } from '../containers/Login/';


describe ('LOGIN', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<Login />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const wrapper = shallow(<Login />)
    const expected = {email: '', passwordOne: '', signUp: false, locationType: '', businessName: '', address: '', phoneNumber: '', contactName: '', foodType: '', logo: null };
    expect(wrapper.state()).toEqual(expected);
  });

  // it('calls handleSubmit onSubmit of the form', () => {
  //   const wrapper = mount(<Login />)
  //   const spy = jest.spyOn(wrapper.instance(), 'handleSubmit');
  //   const mockEvent = { preventDefault: jest.fn() }
  //   wrapper.instance().forceUpdate();
  //   wrapper.find('form').simulate('submit', mockEvent)
  //   expect(spy).toHaveBeenCalled()
  // })

  // describe('mapStateToProps', () => {
  //   it('should have access to currentUser', () => {
  //     const mockStore = {
       
  //     }
  //     const expected = {...mockStore};
  //     const result = mapStateToProps(mockStore);
  //     expect(result).toEqual(expected);
  //   });
  // });

  // describe('mapDispatchToProps', () => {
  //   it('should call dispatch when loadUser is invoked', () => {
  //     const mockDispatch = jest.fn();
  //     const mappedProps = mapDispatchToProps(mockDispatch);
  //     mappedProps.loadUser(mockUser);
  //     expect(mockDispatch).toHaveBeenCalled();
  //   });
  // });
});
