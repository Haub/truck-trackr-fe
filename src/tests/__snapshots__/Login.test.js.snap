// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOGIN should match the snapshot 1`] = `
<main
  className="login"
>
  <form
    className="login-form"
    onSubmit={[Function]}
  >
    <input
      className="email-input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="email"
    />
    <input
      className="password-input"
      name="passwordOne"
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <input
      className="hidden"
      name="businessName"
      onChange={[Function]}
      placeholder="Business Name"
    />
    <input
      className="hidden"
      name="address"
      onChange={[Function]}
      placeholder="Address/City Located"
    />
    <input
      className="hidden"
      name="phoneNumber"
      onChange={[Function]}
      placeholder="Phone Number"
    />
    <input
      className="hidden"
      name="contactName"
      onChange={[Function]}
      placeholder="Contact Name"
    />
    <input
      className="hidden"
      name="foodType"
      onChange={[Function]}
      placeholder="Food Type"
    />
    <p
      className="hidden"
    >
      What Type of Business:
    </p>
    <div
      className="button-holder"
    >
      <input
        className="food-truck-radio"
        id="food-truck-button"
        name="locationType"
        onChange={[Function]}
        type="radio"
        value="food_trucks"
      />
      <label
        className="food-truck-label"
        htmlFor="food-truck-button"
      >
        Food Truck
      </label>
      <input
        className="brewery-radio"
        id="brewery-button"
        name="locationType"
        onChange={[Function]}
        type="radio"
        value="breweries"
      />
      <label
        className="brewery-label"
        htmlFor="brewery-button"
      >
        Brewery
      </label>
    </div>
    <NavLink
      activeClassName="active"
      aria-current="page"
      className="signin-button"
      onClick={[Function]}
      to="/profile"
    >
      Sign In
    </NavLink>
  </form>
  <button
    className="signup-button"
    onClick={[Function]}
  >
    First Time Here? Sign Up!
  </button>
</main>
`;
